TESTS/netsocket/tcp/main.cpp

```diff
@@ -66,6 +66,24 @@ nsapi_version_t get_ip_version()
     return test.get_ip_version();
 }
 
+bool check_oversized_packets(nsapi_error_t error, int &size)
+{
+    if (error == NSAPI_ERROR_PARAMETER) {
+        if (get_ip_version() == NSAPI_IPv4) {
+            if (size > tcp_global::MAX_SEND_SIZE_IPV4) {
+                size = tcp_global::MAX_SEND_SIZE_IPV4;
```

Comment:
I don't understand what you aim to do by modifying the `size` variable.
The function did not send the data, so you should not claim that it did.

Please refactor this function to take just `int size` parameter, so that it does not modify the original error code.

Reply:
I was trying to make sure that the next loop would not use oversized packet (we want to limit the size to 536, so that the modem accepts it).
Another approach that would perhaps be more obvious is to use the return value somehow. For example return 0 (NSAPI_ERROR_OK) if the packet was not oversized or return the suggested size otherwise?

Reply:
The modem can never reject a large TCP request (unless the TCP connection has broken). The underlying call may send a partial amount, which would be visible if you were using non-blocking. But that's always true.  You just subtract off whatever they sent, then ask to send the remainder.

There should be no need to change the TCP tests specifically for modems with small IP MTUs, as the small IP MTU does not affect the TCP stream API.  You do need to allow for limited UDP payloads.